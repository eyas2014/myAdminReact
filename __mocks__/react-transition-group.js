import React from 'react';

let result=[]
function isCalledWith(){

	return result;
};

function reset(){

	result=[];
};

function CSSTransition(props){
	const {classNames, timeout, unmountOnExit}=props;
	result.push({in: props.in, classNames, timeout, unmountOnExit});
	return (props.children);
}



export {CSSTransition, isCalledWith, reset};