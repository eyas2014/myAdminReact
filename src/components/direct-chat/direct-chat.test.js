import React from 'react';
import DirectChat from './direct-chat.jsx';
import enzyme from 'enzyme';

jest.mock('react-transition-group');
jest.mock('./message.jsx');

var messages=[{
	name: 'Alexander Pierce',
	time: new Date(),
	message: "Is this template really for free? That's unbelievable!"},
	{
	name: 'Sarah Bullock',
	time: new Date(),
	message: "You better believe it!"},
	{
	name: 'Alexander Pierce',
	time: new Date(),
	message: "Working with AdminLTE on a great app! Wanna join?"},
	{
	name: 'Sarah Bullock',
	time: new Date(),
	message: "I would love to."}
];

document.body.innerHTML='<div  id="root"></div>';

const mountWrapper=enzyme.mount(<DirectChat data={messages}></DirectChat>, {attachTo: document.getElementById('root')});

it('should render correct htmls',()=>{
	expect(mountWrapper.html()).toMatchSnapshot();
});

it('should render correct htmls',()=>{
	expect(require('./message.jsx').isCalledWith().length).toEqual(4);
	expect(mountWrapper.find('.wrapper main .message').length).toEqual(4);
});

it('should collapse after clicking toggle', ()=>{
	require('react-transition-group').reset();
	mountWrapper.find('.direct-chat header').childAt(1).childAt(1).simulate('click');
	expect(require('react-transition-group').isCalledWith()[0].classNames).toEqual('componentShutdown');
	expect(require('react-transition-group').isCalledWith()[0].in).toEqual(true);
	expect(require('react-transition-group').isCalledWith()[1].classNames).toEqual('componentCollapse');
	expect(require('react-transition-group').isCalledWith()[1].in).toEqual(false);
});

it('should expand after clicking toggle', ()=>{
	require('react-transition-group').reset();
	mountWrapper.find('.direct-chat header').childAt(1).childAt(1).simulate('click');
	expect(require('react-transition-group').isCalledWith()[0].classNames).toEqual('componentShutdown');
	expect(require('react-transition-group').isCalledWith()[0].in).toEqual(true);
	expect(require('react-transition-group').isCalledWith()[1].classNames).toEqual('componentCollapse');
	expect(require('react-transition-group').isCalledWith()[1].in).toEqual(true);
});

it('should close after clicking close', ()=>{
	require('react-transition-group').reset();
	mountWrapper.find('.direct-chat header').childAt(1).childAt(3).simulate('click');
	expect(require('react-transition-group').isCalledWith()[0].classNames).toEqual('componentShutdown');
	expect(require('react-transition-group').isCalledWith()[0].in).toEqual(false);
});
