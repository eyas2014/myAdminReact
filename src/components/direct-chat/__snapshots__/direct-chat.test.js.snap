// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correct htmls 1`] = `"<div class=\\"direct-chat box-shadow\\"><header class=\\"box-border-bottom\\"><div class=\\"btn\\"><span>Direct Chat</span></div><div class=\\"float-right btn-group\\"><div class=\\"btn\\"><span class=\\"bg-warning badge badge-pill text-white font-weight-bold\\">3</span></div><div class=\\"btn\\"><i class=\\"fa fa-minus text-muted\\"></i></div><div class=\\"btn\\"><i class=\\"fas fa-comments text-muted\\"></i></div><div class=\\"btn\\"><i class=\\"fa fa-times text-muted\\"></i></div></div></header><div class=\\"wrapper\\"><main class=\\"messages box-border-bottom\\"><div class=\\"message\\"></div><div class=\\"message\\"></div><div class=\\"message\\"></div><div class=\\"message\\"></div></main><footer class=\\"\\"><div class=\\"input-group p-3\\"><input type=\\"text\\" class=\\"form-control\\" placeholder=\\"tpye message...\\"><span class=\\"input-group-addon bg-warning text-white\\" id=\\"inputGroup-sizing-default\\">Send</span></div></footer></div></div>"`;
